FiniteStateMachine/src/demo/Gearbox.java

```diff
@@ -1,8 +1,8 @@
 package demo;
 
-import statemachine.StateContainer;
+import statemachine.StateMachine;
 
-final class Gearbox extends StateContainer<Gear> {
+final class Gearbox extends StateMachine<Gear> {
```

Comment:
`StateContainer` wasn't a good name, but I'm not sure `StateMachine` encompasses the purpose of this class correctly. Maybe something like `StatefulObject`? idk